modules/apps/29-fee/keeper/msg_server.go

```diff
@@ -25,9 +27,51 @@ func (k Keeper) RegisterCounterpartyAddress(goCtx context.Context, msg *types.Ms
 	return &types.MsgRegisterCounterpartyAddressResponse{}, nil
 }
 
-// EscrowPacketFee defines a rpc handler method for MsgEscrowPacketFee
-// EscrowPacketFee is an open callback that may be called by any module/user that wishes to escrow funds in order to
-// incentivize the relaying of the given packet.
-func (k Keeper) EscrowPacketFee(goCtx context.Context, msg *types.MsgEscrowPacketFee) (*types.MsgEscrowPacketFeeResponse, error) {
-	return &types.MsgEscrowPacketFeeResponse{}, nil
+// PayPacketFee defines a rpc handler method for MsgPayPacketFee
+// PayPacketFee is an open callback that may be called by any module/user that wishes to escrow funds in order to relay the packet with the next sequence
+func (k Keeper) PayPacketFee(goCtx context.Context, msg *types.MsgPayPacketFee) (*types.MsgPayPacketFeeResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// get the next sequence
+	sequence, found := k.GetNextSequenceSend(ctx, msg.SourcePortId, msg.SourceChannelId)
```

Comment:
@AdityaSripal Is this what you had in mind for getting the sequence of the packet for the case whereby we want to pay for a packet that has a bundled 29-fee message + 20-transfer message?

Reply:
Yes, I believe done this way it requires that the fee msg is placed first in the multimsg tx. This should be well-documented